### ISSUE TITLE
Grafana-toolkit has a static output directory

### ISSUE THREAD

**Issue by danmx** [grafana/grafana#37478]
**What happened**:

I'm trying to integrate `grafana-toolkit` with [Bazel](https://bazel.build) and it's hard (impossible?) because I can't override`dist` output directory (all Bazel examples rely on that)

**What you expected to happen**:

I would expect to use features of `npm` to configure my custom output directory. 

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7/8
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

**Comment by ying-jeanne** [comment]
Hi team, @grafana/plugins-platform user is asking to make the output directory of grafana-toolkit configurable, It seems like a feature request, could you take a look? Thanks

**Comment by jackw** [comment]
@danmx I'm afraid I'm not familiar with Bazel. Could you provide a bit more info (or maybe an example repo) on how you're  integrating `grafana-toolkit` with Bazel to help us reproduce the issue you're facing please?

**Comment by cfife-btig** [comment]
Hey @danmx, I know this issue is a bit old but I was having the exact same frustration. If you are pretty up-to-date with your bazel nodejs rules (i.e. maybe 4.7 or 5.2), they recently added the ability to pass dynamic path values into environment variables for build rules. Grafana-Toolkit also supports custom webpack configs.  So one way potential workaround is to copy the `dist` folder into your desired output directory in webpack's config using environment variables. 

This isn't a super simple solution, but it's the best thing that I could get to work. 

First, add a webpack.config.js file (you will also need a folder copying library - like fs-extra)

```
const fse = require('fs-extra');
                              
module.exports = (defaultConfig, options) => {
  defaultConfig.plugins.push(function() {
    this.plugin('done', () => {
      fse.copySync('dist', process.env.OUTPUT_PATH)
    })
  })
  return defaultConfig;
}
```

Then, set the environment variable OUTPUT_PATH using the `env` attribute. Note that the output path is relative to the workspace root, so if your plugin's path is `/grafana/grafana-plugins/my-plugin`, the `OUTPUT_PATH` should be set to `../../../$(@D)` (basically keep going backwards in your path until you reach your workspace file).   

`$(@D)` will become `bazel-out/<your_build_env>/bin/grafana-plugins/<my-plugin>/<my_rule_name>`

```
load("@npm//@grafana/toolkit:index.bzl", "grafana_toolkit")

package(default_visibility = ["//visibility:public"])

APPLICATION_DEPS = glob(
    [
        "**/*",
    ],
)

grafana_toolkit(
    name = "dev",
    args = [
        "plugin:dev",
    ],
    chdir = package_name(),
    data = APPLICATION_DEPS + [
        "@npm//@grafana/data",
        "@npm//@grafana/toolkit",
        "@npm//@grafana/ui",
        "@npm//emotion",
        "@npm//fs-extra",
    ],
    env = {
        "OUTPUT_PATH": "$(@D)",     # This is the important line of code 
    },
    output_dir = True,
)

```

One last tip, If you aren't up to date with rules_nodejs 4.7 or 5.2, what you could also do potentially is just rename your build rule to `dist` and then run it in the bazel-output directory. Use the attribute `chdir = "$(ruledir)"` to run in `bazel-out`. Use the `js_library` rule on all your source code to put it all inside the bazel-out directory. This kinda sucks because you have to name your rule `dist`, but that's the other way I got this to work.

---

For @jackw, honestly, Bazel is a pretty complicated and dense tool. Basically, Bazel + bazel/node tools generally are not very flexible and they don't have a way to easily know which path to retrieve output from (i.e. the `/dist` folder). Instead, it's much much easier to tell a binary/executable (i.e. grafana-toolkit) where to put it's output rather than telling bazel how or where to find it. In my honest opinion, this more bazel/node people's problem than grafana's. But if there was a way to specify an output path, that would make grafana a lot easier to work with Bazel (i.e. not doing the work-around I wrote above).


Basically, if it was possible to do something like this

```
grafana-toolkit plugin:build --outputPath=<your_desired_path>
```

Where `your_desired_path` would default to `dist` but could potentially be any path, that would make things a lot easier for anyone trying to integrate grafana-toolkit with Bazel.

**Comment by jackw** [comment]
I'm still not sure about Bazel but you can customise the output path by creating a `webpack.config.js` in your plugin root folder and add the following:

```js
// webpack.config.js

const path = require('path');

module.exports.getWebpackConfig = (config, options) => ({
  ...config,
  output: {
    ...config.output,
    path: path.resolve(__dirname, 'bazel', 'dist'), // <-- this is the custom 
  }
});
```
When `@grafana/toolkit` is run it will pick this up and webpack will use the `output.path` defined in the example above as the "dist" location.
